aleemer/Internify
# [IN-54, IN-65] Login Authentication

#### User Story
[[IN-54] Login Authentication](https://chaosneutral.atlassian.net/jira/software/projects/IN/boards/1?selectedIssue=IN-54)
[[IN-65] Sign Out option on Navbar](https://chaosneutral.atlassian.net/jira/software/projects/IN/boards/1?selectedIssue=IN-65)

#### Changes made
- Added Firebase to our project
- Added Firebase Google Login Authentication to both `Landing` and `Login` pages
- Added Firebase Email Login Authentication
- Updated Authenticated Navbar to feature 'Sign Out' option
- Wrapped authenticated routes to `Create`, `Edit`, `Feedback`
- Wrapped unauthenticated routes to `Landing` and `Login`

#### Does your new code introduce new warnings on dev console?
No

#### Test Steps

Google Login Test Steps
1. Run the app
2. Click on `Continue with Google` on the hero banner, a popup should appear
3. Continue the steps and select your Google account to authenticate with
4. You should be directed to the `Create` page
5. Attempt to go to a page that does not exist, you should be redirected to the `Create page`
6. On the `AuthNavbar`, select `Sign Out` to sign out, you should be redirected to the `Landing`page
7. Navigate to the `Login` page, and click on `Continue with Google`, a popup should appear
8. Repeat steps 3-6

**Email Login Test Steps (Sign Up)**

1. Run the app ($ npm start) -> will be directed to "Landing Page"
2. Within the "Hero" component, fill in the following fields (i) Create an account with Email (add your email) (ii) Password (create a password to be associated with that email address)
3. Click on "Continue with Email"
4. You should receive an email from Internify to click a link to verify your email address.  note that verifying email is not yet required for access to rest of site (though this may change). 
5. You should be directed to the 'Create' page
6. On the 'AuthNavBar', select 'Sign Out' to sign out, you should be redirected to the 'Landing Page'
7. Continue to "Email Login Test Steps (Sign in - successful sign in)"


**Email Login Test Steps (Sign in - successful sign in)**

1. You are at the "Landing" page
2. Click on "Sign In" found on the the upper right hand side of the "Landing" page
3. In the "Email" text field, fill in the email address you used in step two of "Email Login Test Steps (Sign Up)" above. 
4. In the "Password" text field, fill in the password you used in step two of "Email Login Test Steps (Sign Up)" above.
5. Click "Login"
6. You should be directed to the 'Create' page
7. On the 'AuthNavBar', select 'Sign Out' to sign out, you should be redirected to the 'Landing Page'
8. Continue to "Email Login Test Steps (Sign in - unsuccessful sign in)"

**Email Login Test Steps (Sign in - unsuccessful sign in)**

1. You are at the "Landing" page
2. Click on "Sign In" found on the the upper right hand side of the "Landing" page
3. In the "Email" text field, make up an email that you have never used before. 
4. In the "Password" text field, make up a password that you have never used before.
5. Click "Login"
6. You should receive an alert notifying you that there is something wrong with the email and password combination you have submitted.
